bug: cannot get deterministic model for testing with `get_model`

```
   def __init__(self, **kwargs):
        """
        Initialize with a list of outputs to return in sequence.
        """
>       self.config = DeterministicModelConfig(**kwargs)
E       TypeError: DeterministicModelConfig.__init__() got an unexpected keyword argument 'model_kwargs'
```

This is probably because we do some setdefault with `model_kwargs`

--- a/src/minisweagent/models/__init__.py
+++ b/src/minisweagent/models/__init__.py
@@ -50,14 +50,12 @@ def get_model(input_model_name: str | None = None, config: dict | None = None) -
     config = copy.deepcopy(config)
     config["model_name"] = resolved_model_name
 
-    # API key resolution (from env -> config -> None)
-    if "model_kwargs" not in config:
-        config["model_kwargs"] = {}
-    if from_env := os.getenv("MSWEA_MODEL_API_KEY"):
-        config["model_kwargs"]["api_key"] = from_env
+    model_class = get_model_class(resolved_model_name, config.pop("model_class", ""))
 
-    model_class = config.pop("model_class", "")
-    return get_model_class(resolved_model_name, model_class)(**config)
+    if from_env := os.getenv("MSWEA_MODEL_API_KEY") and not str(type(model_class)).endswith("DeterministicModel"):
+        config.setdefault("model_kwargs", {})["api_key"] = from_env
+
+    return model_class(**config)
 
 
 def get_model_name(input_model_name: str | None = None, config: dict | None = None) -> str:
@@ -76,6 +74,7 @@ def get_model_name(input_model_name: str | None = None, config: dict | None = No
 _MODEL_CLASS_MAPPING = {
     "anthropic": "minisweagent.models.anthropic.AnthropicModel",
     "litellm": "minisweagent.models.litellm_model.LitellmModel",
+    "deterministic": "minisweagent.models.test_models.DeterministicModel",
 }
 
 

--- a/src/minisweagent/models/__init__.py
+++ b/src/minisweagent/models/__init__.py
@@ -52,7 +52,7 @@ def get_model(input_model_name: str | None = None, config: dict | None = None) -
 
     model_class = get_model_class(resolved_model_name, config.pop("model_class", ""))
 
-    if from_env := os.getenv("MSWEA_MODEL_API_KEY") and not str(type(model_class)).endswith("DeterministicModel"):
+    if (from_env := os.getenv("MSWEA_MODEL_API_KEY")) and not str(type(model_class)).endswith("DeterministicModel"):
         config.setdefault("model_kwargs", {})["api_key"] = from_env
 
     return model_class(**config)

--- a/tests/models/test_init.py
+++ b/tests/models/test_init.py
@@ -103,85 +103,39 @@ class TestGetModel:
     def test_env_var_overrides_config_api_key(self):
         """Test that MSWEA_MODEL_API_KEY overrides config api_key."""
         with patch.dict(os.environ, {"MSWEA_MODEL_API_KEY": "env-key"}):
-            # Capture the arguments passed to the model constructor
-            captured_kwargs = {}
+            config = {"model_kwargs": {"api_key": "config-key"}, "model_class": "litellm"}
+            model = get_model("test-model", config)
 
-            class MockModel:
-                def __init__(self, **kwargs):
-                    captured_kwargs.update(kwargs)
-
-            def mock_model_constructor(**kwargs):
-                return MockModel(**kwargs)
-
-            with patch("minisweagent.models.get_model_class") as mock_get_class:
-                mock_get_class.return_value = mock_model_constructor
-
-                config = {"model_kwargs": {"api_key": "config-key"}}
-                get_model("test-model", config)
-
-                assert captured_kwargs["model_kwargs"]["api_key"] == "env-key"
+            # LitellmModel stores the api_key in model_kwargs
+            assert model.config.model_kwargs["api_key"] == "env-key"
 
     def test_config_api_key_used_when_no_env_var(self):
         """Test that config api_key is used when env var is not set."""
         with patch.dict(os.environ, {}, clear=True):
-            # Capture the arguments passed to the model constructor
-            captured_kwargs = {}
+            config = {"model_kwargs": {"api_key": "config-key"}, "model_class": "litellm"}
+            model = get_model("test-model", config)
 
-            class MockModel:
-                def __init__(self, **kwargs):
-                    captured_kwargs.update(kwargs)
-
-            def mock_model_constructor(**kwargs):
-                return MockModel(**kwargs)
-
-            with patch("minisweagent.models.get_model_class") as mock_get_class:
-                mock_get_class.return_value = mock_model_constructor
-
-                config = {"model_kwargs": {"api_key": "config-key"}}
-                get_model("test-model", config)
-
-                assert captured_kwargs["model_kwargs"]["api_key"] == "config-key"
+            # LitellmModel stores the api_key in model_kwargs
+            assert model.config.model_kwargs["api_key"] == "config-key"
 
     def test_env_var_sets_api_key_when_no_config_key(self):
         """Test that MSWEA_MODEL_API_KEY is used when config has no api_key."""
         with patch.dict(os.environ, {"MSWEA_MODEL_API_KEY": "env-key"}):
-            # Capture the arguments passed to the model constructor
-            captured_kwargs = {}
+            config = {"model_class": "litellm"}
+            model = get_model("test-model", config)
 
-            class MockModel:
-                def __init__(self, **kwargs):
-                    captured_kwargs.update(kwargs)
-
-            def mock_model_constructor(**kwargs):
-                return MockModel(**kwargs)
-
-            with patch("minisweagent.models.get_model_class") as mock_get_class:
-                mock_get_class.return_value = mock_model_constructor
-
-                config = {}
-                get_model("test-model", config)
-                assert captured_kwargs["model_kwargs"]["api_key"] == "env-key"
+            # LitellmModel stores the api_key in model_kwargs
+            assert model.config.model_kwargs["api_key"] == "env-key"
 
     def test_no_api_key_when_none_provided(self):
         """Test that no api_key is set when neither env var nor config provide one."""
         with patch.dict(os.environ, {}, clear=True):
-            # Capture the arguments passed to the model constructor
-            captured_kwargs = {}
+            config = {"model_class": "litellm"}
+            model = get_model("test-model", config)
 
-            class MockModel:
-                def __init__(self, **kwargs):
-                    captured_kwargs.update(kwargs)
-
-            def mock_model_constructor(**kwargs):
-                return MockModel(**kwargs)
-
-            with patch("minisweagent.models.get_model_class") as mock_get_class:
-                mock_get_class.return_value = mock_model_constructor
-
-                config = {}
-                get_model("test-model", config)
-                model_kwargs = captured_kwargs.get("model_kwargs", {})
-                assert "api_key" not in model_kwargs
+            # LitellmModel should not have api_key when none provided
+            model_kwargs = getattr(model.config, "model_kwargs", {})
+            assert "api_key" not in model_kwargs
 
     def test_get_deterministic_model(self):
         """Test that get_model can instantiate DeterministicModel via model_class parameter."""

--- a/tests/models/test_init.py
+++ b/tests/models/test_init.py
@@ -106,17 +106,17 @@ class TestGetModel:
             # Capture the arguments passed to the model constructor
             captured_kwargs = {}
 
+            class MockModel:
+                def __init__(self, **kwargs):
+                    captured_kwargs.update(kwargs)
+
             def mock_model_constructor(**kwargs):
-                captured_kwargs.update(kwargs)
-                return DeterministicModel(
-                    outputs=kwargs.get("outputs", ["test"]),
-                    model_name=kwargs.get("model_name", "test"),
-                )
+                return MockModel(**kwargs)
 
             with patch("minisweagent.models.get_model_class") as mock_get_class:
                 mock_get_class.return_value = mock_model_constructor
 
-                config = {"model_kwargs": {"api_key": "config-key"}, "outputs": ["test"]}
+                config = {"model_kwargs": {"api_key": "config-key"}}
                 get_model("test-model", config)
 
                 assert captured_kwargs["model_kwargs"]["api_key"] == "env-key"
@@ -127,17 +127,17 @@ class TestGetModel:
             # Capture the arguments passed to the model constructor
             captured_kwargs = {}
 
+            class MockModel:
+                def __init__(self, **kwargs):
+                    captured_kwargs.update(kwargs)
+
             def mock_model_constructor(**kwargs):
-                captured_kwargs.update(kwargs)
-                return DeterministicModel(
-                    outputs=kwargs.get("outputs", ["test"]),
-                    model_name=kwargs.get("model_name", "test"),
-                )
+                return MockModel(**kwargs)
 
             with patch("minisweagent.models.get_model_class") as mock_get_class:
                 mock_get_class.return_value = mock_model_constructor
 
-                config = {"model_kwargs": {"api_key": "config-key"}, "outputs": ["test"]}
+                config = {"model_kwargs": {"api_key": "config-key"}}
                 get_model("test-model", config)
 
                 assert captured_kwargs["model_kwargs"]["api_key"] == "config-key"
@@ -148,17 +148,17 @@ class TestGetModel:
             # Capture the arguments passed to the model constructor
             captured_kwargs = {}
 
+            class MockModel:
+                def __init__(self, **kwargs):
+                    captured_kwargs.update(kwargs)
+
             def mock_model_constructor(**kwargs):
-                captured_kwargs.update(kwargs)
-                return DeterministicModel(
-                    outputs=kwargs.get("outputs", ["test"]),
-                    model_name=kwargs.get("model_name", "test"),
-                )
+                return MockModel(**kwargs)
 
             with patch("minisweagent.models.get_model_class") as mock_get_class:
                 mock_get_class.return_value = mock_model_constructor
 
-                config = {"outputs": ["test"]}
+                config = {}
                 get_model("test-model", config)
                 assert captured_kwargs["model_kwargs"]["api_key"] == "env-key"
 
@@ -168,17 +168,17 @@ class TestGetModel:
             # Capture the arguments passed to the model constructor
             captured_kwargs = {}
 
+            class MockModel:
+                def __init__(self, **kwargs):
+                    captured_kwargs.update(kwargs)
+
             def mock_model_constructor(**kwargs):
-                captured_kwargs.update(kwargs)
-                return DeterministicModel(
-                    outputs=kwargs.get("outputs", ["test"]),
-                    model_name=kwargs.get("model_name", "test"),
-                )
+                return MockModel(**kwargs)
 
             with patch("minisweagent.models.get_model_class") as mock_get_class:
                 mock_get_class.return_value = mock_model_constructor
 
-                config = {"outputs": ["test"]}
+                config = {}
                 get_model("test-model", config)
                 model_kwargs = captured_kwargs.get("model_kwargs", {})
                 assert "api_key" not in model_kwargs

--- a/tests/models/test_init.py
+++ b/tests/models/test_init.py
@@ -183,6 +183,16 @@ class TestGetModel:
                 model_kwargs = captured_kwargs.get("model_kwargs", {})
                 assert "api_key" not in model_kwargs
 
+    def test_get_deterministic_model(self):
+        """Test that get_model can instantiate DeterministicModel via model_class parameter."""
+        config = {"outputs": ["hello", "world"], "cost_per_call": 2.0}
+        model = get_model("test-model", config | {"model_class": "deterministic"})
+
+        assert isinstance(model, DeterministicModel)
+        assert model.config.outputs == ["hello", "world"]
+        assert model.config.cost_per_call == 2.0
+        assert model.config.model_name == "test-model"
+
 
 class TestGlobalModelStats:
     def test_prints_cost_limit_when_set(self, capsys):

